sql/src/main/java/io/crate/operation/aggregation/impl/SumAggregation.java

```diff
@@ -31,57 +31,103 @@
 import io.crate.types.DataTypes;
 import org.elasticsearch.common.breaker.CircuitBreakingException;
 
+import javax.annotation.Nullable;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
 
-public class SumAggregation extends AggregationFunction<Double, Double> {
+public class SumAggregation<T extends Number> extends AggregationFunction<T, T> {
 
     public static final String NAME = "sum";
 
     private final FunctionInfo info;
+    private final BiFunction<T, T, T> addition;
+    private final DataType<T> type;
+    private final int bytesSize;
+
+    public static final Map<DataType, SumAggregation> sumFunctions;
```

Comment:
This seems to only be public - and only have the DataType as key because of a benchmark. I'd be in favor  of creating the `Functions` registry in the benchmark and avoid this special testing code. The map is otherwise not required